pkg/injector/envoy_container.go

```diff
@@ -49,11 +52,18 @@ func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator
 		ImagePullPolicy: corev1.PullIfNotPresent,
 		SecurityContext: securityContext,
 		Ports:           getEnvoyContainerPorts(originalHealthProbes),
-		VolumeMounts: []corev1.VolumeMount{{
-			Name:      envoyBootstrapConfigVolume,
-			ReadOnly:  true,
-			MountPath: envoyProxyConfigPath,
-		}},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      envoyBootstrapConfigVolume,
+				ReadOnly:  true,
+				MountPath: envoyProxyConfigPath,
+			},
+			{
+				Name:      envoyXDSSecretVolume,
+				ReadOnly:  false,
+				MountPath: envoyXDSSecretPath,
```

Comment:
I agree that it would be simpler to keep everything in one secret.  With the changes in this PR, the `bootstrap.yaml` no longer contains any secret information and therefore is better suited for a ConfigMap. However, I'm not sure it is worth it to create an additional resource for every injected Pod. I'm not sure how important it is to only store data that is intended to be secret in a Secret. 
@steeling and @keithmattix could you share your thoughts on this?

Reply:
Yup, I think the short answer is we need to get some feedback from customers on how they want to use this and the [keyvault feature](#4501 ). From what I've been told the requirement is that some customers can't use k8s secrets. In which case, we need to store them elsewhere.

Splitting them out gives us the most flexibility on where "elsewhere" is, in that it allows us to use both an encrypted key vaule store (in which case we could keep them joined), but also some PKI (in which case they would need to be separate).

@trstringer, @shashankram can you comment on whether this is true for the given key vault requirements?

Reply:
Spoke with @steeling offline. I am going to move forward with keeping the bootstrap config and certificates in one k8s Secret. I will update the PR accordingly. Separating the certificates into their own k8s Secret for the reasons mentioned above can be revisited at a later time